client/web/web.go

```diff
@@ -262,10 +270,52 @@ func (s *Server) serveAPI(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, "method not allowed", http.StatusMethodNotAllowed)
 		}
 		return
+	case strings.HasPrefix(path, "/local/"):
+		s.proxyRequestToLocalAPI(w, r)
+		return
 	}
 	http.Error(w, "invalid endpoint", http.StatusNotFound)
 }
 
+// verifyRequestHost checks the request header to verify that the request
+// came from the node's web client frontend, and not some other source.
```

Comment:
this is not exactly true - the request could've come from any web client that can reach this web UI, not just our official frontend.
maybe update to something like `verifyRequestHost checks that the request was intended for a known web client address. This is used to prevent DNS rebinding attacks, where the browser is tricked into opening the web client by some other name.`